edk2toolext/versioninfo/versioninfo_helper.py

```diff
@@ -0,0 +1,556 @@
+VAR_FILE_INFO_STR = "VarFileInfo"
+
+# Key values for interacting with pefile objects
+PE_ENCODING = "utf-8"
+PE_STRUCT_STR = "Structure"
+PE_VALUE_STR = "Value"
+BEGIN_STR = "BEGIN"
+END_STR = "END"
+BLOCK_STR = "BLOCK"
+VALUE_STR = "VALUE"
+
+# Validation requirements
+VERSIONFILE_REQUIRED_FIELDS = {
+    FILE_VERSION_STR,
+    PRODUCT_VERSION_STR,
+    FILE_FLAG_MASK_STR,
+    FILE_FLAGS_STR,
+    FILE_OS_STR,
+    FILE_TYPE_STR,
+    FILE_SUBTYPE_STR,
+    STRING_FILE_INFO_STR,
+    VAR_FILE_INFO_STR
+}
+
+VERSIONFILE_ALLOWED_FIELDS = {
+    FILE_VERSION_STR,
+    PRODUCT_VERSION_STR,
+    FILE_FLAG_MASK_STR,
+    FILE_FLAGS_STR,
+    FILE_OS_STR,
+    FILE_TYPE_STR,
+    FILE_SUBTYPE_STR,
+    STRING_FILE_INFO_STR,
+    VAR_FILE_INFO_STR,
+    SIGNATURE_STR,
+    STRUC_VERSION_STR,
+    FILE_DATE_STR,
+    FILE_DATE_MS_STR,
+    FILE_DATE_LS_STR
+}
+
+COMPANY_NAME_STR = "CompanyName"
+PRODUCT_NAME_STR = "ProductName"
+
+STRING_FILE_INFO_REQUIRED_FIELDS = {
+    COMPANY_NAME_STR,
+    PRODUCT_NAME_STR,
+    FILE_VERSION_STR
+}
+
+STRING_FILE_INFO_ALLOWED_FIELDS = {
+    "Comments",
+    "CompanyName",
+    "FileDescription",
+    "FileVersion",
+    "InternalName",
+    "LegalCopyright",
+    "LegalTrademarks",
+    "OriginalFilename",
+    "PrivateBuild",
+    "ProductName",
+    "ProductVersion",
+    "SpecialBuild"
+}
+
+VALID_FILE_OS_VALUES = {
+    "VOS_UNKNOWN",
+    "VOS_DOS",
+    "VOS_NT",
+    "VOS__WINDOWS16",
+    "VOS__WINDOWS32",
+    "VOS_DOS_WINDOWS16",
+    "VOS_DOS_WINDOWS32",
+    "VOS_NT_WINDOWS32"
+}
+
+VALID_FILE_TYPE_VALUES = {
+    "VFT_APP",
+    "VFT_DLL",
+    "VFT_DRV",
+    "VFT_FONT",
+    "VFT_STATIC_LIB",
+    "VFT_UNKNOWN",
+    "VFT_VXD"
+}
+
+VALID_SUBTYPE_VFT_DRV = {
+    "VFT2_UNKNOWN",
+    "VFT2_DRV_COMM",
+    "VFT2_DRV_PRINTER",
+    "VFT2_DRV_KEYBOARD",
+    "VFT2_DRV_LANGUAGE",
+    "VFT2_DRV_DISPLAY",
+    "VFT2_DRV_MOUSE",
+    "VFT2_DRV_NETWORK",
+    "VFT2_DRV_SYSTEM",
+    "VFT2_DRV_INSTALLABLE",
+    "VFT2_DRV_SOUND",
+    "VFT2_DRV_VERSIONED_PRINTER"
+}
+
+VALID_SUBTYPE_VFT_FONT = {
+    "VFT2_UNKNOWN",
+    "VFT2_FONT_RASTER",
+    "VFT2_FONT_VECTOR",
+    "VFT2_FONT_TRUETYPE"
+}
+
+VALID_LANG_ID = {
+    0x0401, 0x0402,
+    0x0403, 0x0404,
+    0x0405, 0x0406,
+    0x0407, 0x0408,
+    0x0409, 0x040A,
+    0x040B, 0x040C,
+    0x040D, 0x040E,
+    0x040F, 0x0410,
+    0x0411, 0x0412,
+    0x0413, 0x0414,
+    0x0415, 0x0416,
+    0x0417, 0x0418,
+    0x0419, 0x041A,
+    0x041B, 0x041C,
+    0x041D, 0x041E,
+    0x041F, 0x0420,
+    0x0421, 0x0804,
+    0x0807, 0x0809,
+    0x080A, 0x080C,
+    0x0C0C, 0x100C,
+    0x0816, 0x081A,
+    0x0810, 0x0813,
+    0x0814
+}
+
+VALID_CHARSET_ID = {
+    0x0000, 0x03A4,
+    0x03B5, 0x03B6,
+    0x04B0, 0x04E2,
+    0x04E3, 0x04E4,
+    0x04E5, 0x04E6,
+    0x04E7, 0x04E8
+}
+
+
+class PEObject(object):
+    pe = None
+
+    # Takes filepath to PE/PE+ file to parse
+    def __init__(self, filepath):
+        try:
+            print(filepath)
```

Comment:
should use logging instead of print so that it is captured inside all logs rather than just command line console